onnxruntime/core/optimizer/optimizer_execution_frame.cc

```diff
@@ -119,18 +122,40 @@ OptimizerExecutionFrame::Info::Info(const std::vector<const Node*>& nodes,
   node_index_info_ = std::make_unique<NodeIndexInfo>(nodes, ort_value_name_idx_map_);
 }
 
-Status OptimizerExecutionFrame::Info::TryFindKernel(const Node* node, const KernelCreateInfo** out) const{
+Status OptimizerExecutionFrame::Info::TryFindKernel(const Node* node, const KernelCreateInfo** out) const {
   std::shared_ptr<KernelRegistry> kernel_registry = execution_provider_.GetKernelRegistry();
-  return kernel_registry->TryFindKernel(*node, execution_provider_.Type(), out);
+  const AutoRegisteringKernelTypeStrResolver kernel_type_str_resolver{};
+  return kernel_registry->TryFindKernel(*node, execution_provider_.Type(), kernel_type_str_resolver, out);
```

Comment:
We could, but that would be limited to full build.

Typically, I imagine one would want to re-use an existing resolver if available (I suppose I was too lazy to try to plumb one through here). For AutoRegisteringKernelTypeStrResolver, it could save some redundant registrations.